core/models.py

```diff
@@ -1207,12 +1204,18 @@ def __unicode__(self):
         return "path=%s size=%s sha1=%s" % (self.path, self.size, self.sha1)
 
     def _add_page(self, page, tar):
+        from index import get_page_text
```

Comment:
This belongs at the top of the file

Reply:
No, it doesn't. If it will be at the top of the file a circular import will cause and error

Reply:
One minor note: it'd be better style to use an explicitly relative import in case anything in the search path ever adds a module named `index`: `from .index import get_page_text`.

On a larger note, that would definitely cause a circular import and that suggests that this code might belong somewhere else, especially since this is doing more file I/O than database work and this method and the method which calls it are invoked by exactly one point (the `dump_ocr` task). It seems like it'd be better just to move that code into the task and keep the models clean, which would also keep the import flow consistent.

Reply:
I agree, I am going to create a new ticket for it

Reply:
Why do you need a new ticket? We are expressing concern here, you should leave the code cleaner than it was.

Reply:
When I was reviewing this with Leo, he suggested a separate issue and I agreed because the change is only tangentially related to this work. Since it's small and isolated that could be an argument for a single commit here but I was hoping to keep more of a searchable history (the title on this PR bothers me for the same reasons) with cross-references between the issues.

Reply:
I suppose, I guess I see it more as cleanup than a separate task. I have always taken the liberty of cleaning up code while I am poking around other things. This stems from the fact that when I worked at Boeing we weren't allowed to have tasks just to clean up code, so it was understood you had to practice "leave your code cleaner than you left it", otherwise things would just become unmaintainable.

Reply:
To be clear, I don't see this as an especially strong issue — it _definitely_ should be a separate commit but since we're talking about two methods called in one place & couple dozen lines of code there's a pretty low upper bar for how strongly I'd argue for either option.